### PR TITLE
feat(CRT-146): add sync update logic updating MasterUserRecord in host

### PR DESCRIPTION
**Pull request by MatousJobanek**
depends on https://github.com/codeready-toolchain/toolchain-common/pull/10

adds synchronization logic that updates `MasterUserRecord` with the latest `resourceVersion` of the `UserAccount` by setting it to `SyncIndex` in the spec of the `UserAccountEmbedded`

### CODE REVIEW

File: pkg/controller/useraccountstatus/useraccount_status_controller.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+	return nil
+}
+
+type GenerationNotChangedPredicate struct {
+	predicate.GenerationChangedPredicate
+}
+
+func (p GenerationNotChangedPredicate) Update(e event.UpdateEvent) bool {
+	return p.GenerationChangedPredicate.Update(e)
```
[the remark this answers is not in the document]
Comment (alexeykazakov) [reply pkg/controller/useraccountstatus/useraccount_status_controller.go]: Also, can we add a test for `GenerationNotChangedPredicate`?

[the remark this answers is not in the document]
Comment (MatousJobanek) [reply pkg/controller/useraccountstatus/useraccount_status_controller.go]: Ah, I'm an idiot. Let me fix that and do test for it.

[the remark this answers is not in the document]
Comment (MatousJobanek) [reply pkg/controller/useraccountstatus/useraccount_status_controller.go]: I created a new implementation of the predicate: 9223e6b

**Pull request merged**